### PR TITLE
SAK-47291 Gradebook using external tools as Assignments, Tests & Quizzes are not showing i18n titles

### PR DESCRIPTION
**Pull request by josecebe**

### CODE REVIEW

**Comment by josecebe** [comment]
This change will need a conversion script for institutions that were already using gradebook items on any of the next tools: Tests & Quizzes, Assignments, Lesson Builder and Attendance (sakai contrib).

The conversion script is on this sakai reference PR: https://github.com/sakaiproject/sakai-reference/pull/172

**Comment by josecebe** [comment]
After applying this PR, the titles will be always shown on the current user's language:
SAK-47291

**Review by jesusmmp — APPROVED** [review]
LGTM!

File: webservices/cxf/src/java/org/sakaiproject/webservices/Assignments.java [hunk 1]
```diff
@@ -522,7 +522,7 @@ protected void integrateGradebook( String assignmentRef, String associateGradebo
                                         newTitle,
                                         newAssignment_maxPoints/10,
                                         new Date(newAssignment_dueTime.getTime()),
-                                        "Assignment",
+                                        "sakai.assignment.grades",
```
Comment (adrianfish) [inline webservices/cxf/src/java/org/sakaiproject/webservices/Assignments.java:525]: You could  use AssignmentServiceConstants.ASSIGNMENT_TOOL_ID here.

File: webservices/cxf/src/java/org/sakaiproject/webservices/Assignments.java [hunk 2]
```diff
@@ -494,7 +494,7 @@ protected void integrateGradebook( String assignmentRef, String associateGradebo
                             newAssignment_title,
                             newAssignment_maxPoints/10,
                             new Date(newAssignment_dueTime.getTime()),
-                            "Assignment",
+                            "sakai.assignment.grades",
```
Comment (adrianfish) [inline webservices/cxf/src/java/org/sakaiproject/webservices/Assignments.java:497]: You could use AssignmentServiceConstants.ASSIGNMENT_TOOL_ID here.

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java [hunk 3]
```diff
@@ -3059,10 +3066,10 @@ public String getIconClass(final Assignment assignment) {
 	public Map<String, String> getIconClassMap() {
 		final Map<String, String> mapping = new HashMap<>();
 
-		mapping.put(this.toolManager.getLocalizedToolProperty("sakai.assignment", "title"), getAssignmentsIconClass());
-		mapping.put(this.toolManager.getLocalizedToolProperty("sakai.samigo", "title"), getSamigoIconClass());
-		mapping.put("Lesson Builder", getLessonBuilderIconClass());
-		mapping.put("Attendance", getAttendanceIconClass());
+		mapping.put("sakai.assignment.grades", getAssignmentsIconClass());
```
Comment (adrianfish) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:3069]: You could use AssignmentServiceConstants.ASSIGNMENT_TOOL_ID here.

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java [hunk 4]
```diff
@@ -3036,17 +3037,23 @@ public boolean isValidNumericGrade(String grade)
...
-		} else if (StringUtils.equals(externalAppName, this.toolManager.getLocalizedToolProperty("sakai.samigo", "title"))) {
-			iconClass = getSamigoIconClass();
-		// "Lesson Builder" is currently hardcoded in SimplePageBean.java (no localization required)
-		} else if (StringUtils.equals(externalAppName, "Lesson Builder")) {
-			iconClass = getLessonBuilderIconClass();
-		} else if (StringUtils.equals(externalAppName, "Attendance")) {
-			iconClass = getAttendanceIconClass();
+		String iconClass;
+		switch (externalAppName) {
+			case "sakai.assignment.grades":
```
Comment (adrianfish) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:3042]: You could use AssignmentServiceConstants.ASSIGNMENT_TOOL_ID here.

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 5]
```diff
@@ -3490,7 +3490,7 @@ private void currentAssignmentGradebookIntegrationIntoContext(Context context, S
         // filtering out those from Samigo
         for (Iterator i = gradebookAssignments.iterator(); i.hasNext(); ) {
             org.sakaiproject.grading.api.Assignment gAssignment = (org.sakaiproject.grading.api.Assignment) i.next();
-            if (!gAssignment.getExternallyMaintained() || gAssignment.getExternallyMaintained() && gAssignment.getExternalAppName().equals(assignmentService.getToolTitle())) {
+            if (!gAssignment.getExternallyMaintained() || gAssignment.getExternallyMaintained() && gAssignment.getExternalAppName().equals(assignmentService.getToolId())) {
```
Comment (adrianfish) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:3493]: Will this need a db upgrade script?

**Review by mpellicer — APPROVED** [review]

**Pull request merged**